packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -646,43 +678,157 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
+  ///
+  /// ```dart imports
+  /// import 'package:flutter/foundation.dart';
+  /// ```
+  ///
+  /// ```dart
+  /// UnfocusDisposition disposition = UnfocusDisposition.scope;
+  ///
+  /// @override
+  /// Widget build(BuildContext context) {
+  ///   return Material(
+  ///     child: Container(
+  ///       color: Colors.white,
+  ///       child: Column(
+  ///         mainAxisAlignment: MainAxisAlignment.center,
+  ///         children: <Widget>[
+  ///           Wrap(
+  ///             children: List<Widget>.generate(4, (int index) {
+  ///               return SizedBox(
+  ///                 width: 200,
+  ///                 child: Padding(
+  ///                   padding: const EdgeInsets.all(8.0),
+  ///                   child: TextField(
+  ///                     decoration: InputDecoration(border: OutlineInputBorder()),
+  ///                   ),
+  ///                 ),
+  ///               );
+  ///             }),
+  ///           ),
+  ///           Row(
+  ///             mainAxisAlignment: MainAxisAlignment.spaceAround,
+  ///             children: <Widget>[
+  ///               ...List<Widget>.generate(UnfocusDisposition.values.length,
+  ///                   (int index) {
+  ///                 return Row(
+  ///                   mainAxisSize: MainAxisSize.min,
+  ///                   children: <Widget>[
+  ///                     Radio<UnfocusDisposition>(
+  ///                       groupValue: disposition,
+  ///                       onChanged: (UnfocusDisposition value) {
+  ///                         setState(() {
+  ///                           disposition = value;
+  ///                         });
+  ///                       },
+  ///                       value: UnfocusDisposition.values[index],
+  ///                     ),
+  ///                     Text(describeEnum(UnfocusDisposition.values[index])),
+  ///                   ],
+  ///                 );
+  ///               }),
+  ///               OutlineButton(
+  ///                 child: const Text('UNFOCUS'),
+  ///                 onPressed: () {
+  ///                   setState(() {
+  ///                     primaryFocus.unfocus(disposition: disposition);
+  ///                   });
+  ///                 },
+  ///               ),
+  ///             ],
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     ),
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  void unfocus({
+    UnfocusDisposition disposition = UnfocusDisposition.scope,
+  }) {
+    assert(disposition != null);
     if (!hasFocus && (_manager == null || _manager._markedForFocus != this)) {
       return;
     }
-    if (!hasPrimaryFocus) {
-      // If we are in the focus chain, but not the primary focus, then unfocus
-      // the primary instead.
-      _manager?.primaryFocus?.unfocus(focusPrevious: focusPrevious);
+    FocusScopeNode scope = enclosingScope;
+    if (scope == null) {
+      // Attempting to unfocus the root scope isn't allowed.
```

Comment:
should we assert instead?

Reply:
No, I think it's fine to do nothing when attempting to unfocus the root node.  The fact that the scope is null indicates that this is either a node that isn't part of the tree yet, or that it's the root node, and in either case just doing nothing is the right thing to do. I reworded the comment a bit.